src/FileWriterTask.cpp

```diff
@@ -25,130 +23,142 @@ json hdf_parse(std::string const &Structure) {
     throw FileWriter::ParseError(Structure);
   }
 }
-}
+} // namespace
 
 std::atomic<uint32_t> n_FileWriterTask_created{0};
 
-std::vector<DemuxTopic> &FileWriterTask::demuxers() { return _demuxers; }
+std::vector<DemuxTopic> &FileWriterTask::demuxers() { return Demuxers; }
+
+/// Helper function for creating an ID.
+///
+/// \param ExtraValue Used to help make a unique ID.
+/// \return A "unique" id.
+uint64_t createId(int ExtraValue) {
+  using namespace std::chrono;
+  return (static_cast<uint64_t>(
+              duration_cast<nanoseconds>(system_clock::now().time_since_epoch())
+                  .count())
+          << 16) +
+         (ExtraValue & 0xffff);
+}
 
 FileWriterTask::FileWriterTask(
-    std::string ServiceID_,
+    std::string const &ServiceID_,
```

Comment:
Asking for a const ref here does, against popular belief, not result in performance improvement (which would be for this part of code irrelevant anyway).

On the contrary, by asking for a const ref, you hide for the reader of the API how the value will be used.

In the constructor, we'll make a copy anyway and keep it as member in the class. Therefore, we should take it as a copy already in the constructor's signature, as it was done before.

Alternatively, you could argue that you really want a reference, but then you should also store it as a reference in the class.